src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1821,108 +1809,64 @@ private static ITypeSymbol GetMemberType(ISymbol memberSymbol)
             private IEnumerable<TypeInferenceInfo> InferTypeInRefExpression(RefExpressionSyntax refExpression)
                 => InferTypes(refExpression);
 
-            private IEnumerable<TypeInferenceInfo> InferTypeForReturnStatement(ReturnStatementSyntax returnStatement, SyntaxToken? previousToken = null)
+            private ITypeSymbol UnwrapTaskLike(ITypeSymbol type, bool isAsync)
             {
-                InferTypeForReturnStatement(returnStatement, previousToken,
-                    out var isAsync, out var types);
-
                 if (!isAsync)
                 {
-                    return types;
-                }
-
-                var taskOfT = this.Compilation.TaskOfTType();
-                if (taskOfT == null || types == null)
-                {
-                    return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
+                    return type;
                 }
 
-                return from t in types
-                       where t.InferredType != null && t.InferredType.OriginalDefinition.Equals(taskOfT)
-                       let nt = (INamedTypeSymbol)t.InferredType
-                       where nt.TypeArguments.Length == 1
-                       select new TypeInferenceInfo(nt.TypeArguments[0]);
+                return
+                    type.OriginalDefinition.Equals(this.Compilation.TaskOfTType())
+                        ? ((INamedTypeSymbol)type).TypeArguments[0] :
+                    type.OriginalDefinition.Equals(this.Compilation.TaskType())
+                        ? this.Compilation.GetSpecialType(SpecialType.System_Void) :
+                    type;
```

Comment:
IDE style is either to:
1. align `?` and `:`
2. try to align bodies of `?` and `:`

My preference is 1.

Reply:
If I only have a single `?:`, I do indeed align them but if there are more, I usually make it so that it resembles an if/else chain, so that each condition is on its own line. I'm not quite sure what your desired style would look like in the case of nested conditionals. Can you please give me a code example?

Reply:
either:

```c#
return type.OriginalDefinition.Equals(this.Compilation.TaskOfTType())
    ? ((INamedTypeSymbol)type).TypeArguments[0]
    : type.OriginalDefinition.Equals(this.Compilation.TaskType())
      ? this.Compilation.GetSpecialType(SpecialType.System_Void)
      : type;

// or:

return type.OriginalDefinition.Equals(this.Compilation.TaskOfTType()) ? ((INamedTypeSymbol)type).TypeArguments[0] :
       type.OriginalDefinition.Equals(this.Compilation.TaskType())    ? this.Compilation.GetSpecialType(SpecialType.System_Void) : type;

Reply:
That is becoming quite unreadable. I think I'll just change these to `if`s.

Reply:
'if's work for me as well.

Reply:
➡️ Fixed in 638f14d8a5b36d5415287b4a9774771facc0d2e3